test/jasmine/tests/polygon_test.js

```diff
@@ -0,0 +1,214 @@
+        pts.forEach(function(pt) {
+            expect(poly.contains(pt)).toBe(false);
+            expect(poly.contains(pt, true)).toBe(false);
+            expect(poly.contains(pt, false)).toBe(false);
+        });
+    });
+
+    it('should prepare a polygon object correctly', function() {
+        var polyPts = [squareCW, squareCCW, bowtie, squareish, equilateralTriangle,
+            zigzag, donut, donut2];
+
+        polyPts.forEach(function(polyPt) {
+            var poly = polygonTester(polyPt),
+                xArray = polyPt.map(function(pt) { return pt[0]; }),
+                yArray = polyPt.map(function(pt) { return pt[1]; });
+
+            expect(poly.pts.length).toEqual(polyPt.length + 1);
+            polyPt.forEach(function(pt, i) {
+                expect(poly.pts[i]).toEqual(pt);
+            });
+            expect(poly.pts[poly.pts.length - 1]).toEqual(polyPt[0]);
+            expect(poly.xmin).toEqual(Math.min.apply(null, xArray));
+            expect(poly.xmax).toEqual(Math.max.apply(null, xArray));
+            expect(poly.ymin).toEqual(Math.min.apply(null, yArray));
+            expect(poly.ymax).toEqual(Math.max.apply(null, yArray));
+        });
+    });
+
+    it('should include the whole boundary, except as per omitFirstEdge', function() {
+        var polyPts = [squareCW, squareCCW, bowtie, squareish, equilateralTriangle,
+            zigzag, donut, donut2];
+        var np = 6; // number of intermediate points on each edge to test
+
+        polyPts.forEach(function(polyPt) {
+            var poly = polygonTester(polyPt);
+
+            var isRect = polyPt === squareCW || polyPt === squareCCW;
+            expect(poly.isRect).toBe(isRect);
+            // to make sure we're only using the bounds and first pt, delete the rest
+            if(isRect) poly.pts.splice(1, poly.pts.length);
+
+            poly.pts.forEach(function(pt1, i) {
+                if(!i) return;
+                var pt0 = poly.pts[i - 1],
+                    j;
+
+                var testPts = [pt0, pt1];
+                for(j = 1; j < np; j++) {
+                    if(pt0[0] === pt1[0]) {
+                        testPts.push([pt0[0], pt0[1] + (pt1[1] - pt0[1]) * j / np]);
+                    }
+                    else {
+                        var x = pt0[0] + (pt1[0] - pt0[0]) * j / np;
+                        // calculated the same way as in the pt_in_polygon source,
+                        // so we know rounding errors will apply the same and this pt
+                        // *really* appears on the boundary
+                        testPts.push([x, pt0[1] + (x - pt0[0]) * (pt1[1] - pt0[1]) /
+                                                  (pt1[0] - pt0[0])]);
+                    }
+                }
+                testPts.forEach(function(pt, j) {
+                    expect(poly.contains(pt))
+                        .toBe(true, 'poly: ' + polyPt.join(';') + ', pt: ' + pt);
```

Comment:
the extra arg (in any of the `.toXXX` methods, I think) gives a custom message with failure, so you're not stuck with just "expected false to be true" - really useful when you're testing in a loop!